src/backends/postgresql/statement.cpp

```diff
@@ -23,7 +23,9 @@ namespace // unnamed
 {
 
 // used only with asynchronous operations in single-row mode
-
+#if (__cplusplus >= 201703L)
```

Comment:
It would be better to define `SOCI_MAYBE_UNUSED` once and then use it in both places (and maybe more places later).

Reply:
> The second one blindly fixes the warning instead of actually looking at what it means, namely that these functions are indeed not used when SOCI_POSTGRESQL_NOSINGLEROWMODE is not defined

I didn't see that the other function was also only then not used if the macro was not defined.
I overread the comment above.

> It would be better to define SOCI_MAYBE_UNUSED once and then use it in both places (and maybe more places later).

I do agree, but I am not familiar in your codebase.